src/test/java/org/apache/sysml/test/integration/functions/codegenalg/AlgorithmKMeans.java

```diff
@@ -58,85 +66,245 @@ public void setUp() {
 	}
 	
 	@Test
 	public void testKMeansSparseBinMultiRewritesCP() {
-		runKMeansTest(TEST_NAME1, true, true, 2, 10, ExecType.CP);
+		runKMeansTest(TEST_NAME1, true, true, 2, 10, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansDenseBinMultiCP() {
-		runKMeansTest(TEST_NAME1, false, false, 2, 10, ExecType.CP);
+		runKMeansTest(TEST_NAME1, false, false, 2, 10, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansSparseBinMultiCP() {
-		runKMeansTest(TEST_NAME1, false, true, 2, 10, ExecType.CP);
+		runKMeansTest(TEST_NAME1, false, true, 2, 10, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansDenseMulSingleRewritesCP() {
-		runKMeansTest(TEST_NAME1, true, false, 20, 1, ExecType.CP);
+		runKMeansTest(TEST_NAME1, true, false, 20, 1, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansSparseMulSingleRewritesCP() {
-		runKMeansTest(TEST_NAME1, true, true, 20, 1, ExecType.CP);
+		runKMeansTest(TEST_NAME1, true, true, 20, 1, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansDenseMulSingleCP() {
-		runKMeansTest(TEST_NAME1, false, false, 20, 1, ExecType.CP);
+		runKMeansTest(TEST_NAME1, false, false, 20, 1, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansSparseMulSingleCP() {
-		runKMeansTest(TEST_NAME1, false, true, 20, 1, ExecType.CP);
+		runKMeansTest(TEST_NAME1, false, true, 20, 1, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansDenseMulMultiRewritesCP() {
-		runKMeansTest(TEST_NAME1, true, false, 20, 10, ExecType.CP);
+		runKMeansTest(TEST_NAME1, true, false, 20, 10, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansSparseMulMultiRewritesCP() {
-		runKMeansTest(TEST_NAME1, true, true, 20, 10, ExecType.CP);
+		runKMeansTest(TEST_NAME1, true, true, 20, 10, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansDenseMulMultiCP() {
-		runKMeansTest(TEST_NAME1, false, false, 20, 10, ExecType.CP);
+		runKMeansTest(TEST_NAME1, false, false, 20, 10, ExecType.CP, TestType.DEFAULT);
 	}
 	
 	@Test
 	public void testKMeansSparseMulMultiCP() {
-		runKMeansTest(TEST_NAME1, false, true, 20, 10, ExecType.CP);
+		runKMeansTest(TEST_NAME1, false, true, 20, 10, ExecType.CP, TestType.DEFAULT);
+	}
+
+	@Test
+	public void testKMeansDenseBinSingleRewritesCPFuseAll() {
+		runKMeansTest(TEST_NAME1, true, false, 2, 1, ExecType.CP, TestType.FUSE_ALL);
+	}
+
+	@Test
+	public void testKMeansSparseBinSingleRewritesCPFuseAll() {
+		runKMeansTest(TEST_NAME1, true, true, 2, 1, ExecType.CP, TestType.FUSE_ALL);
+	}
+
+//	@Test
+//	public void testKMeansDenseBinSingleCPFuseAll() {
+//		runKMeansTest(TEST_NAME1, false, false, 2, 1, ExecType.CP, TestType.FUSE_ALL);
+//	}
+//
+//	@Test
+//	public void testKMeansSparseBinSingleCPFuseAll() {
+//		runKMeansTest(TEST_NAME1, false, true, 2, 1, ExecType.CP, TestType.FUSE_ALL);
+//	}
+
```

Comment:
These test cases failed with following error message. Is it not applicable to this case? 

`18/03/08 12:20:57 ERROR api.DMLScript: Failed to execute DML script.
org.apache.sysml.runtime.DMLRuntimeException: org.apache.sysml.runtime.DMLRuntimeException: Failed to compile class codegen.TMP78.
	at org.apache.sysml.hops.codegen.SpoofCompiler.optimize(SpoofCompiler.java:452)
	at org.apache.sysml.hops.recompile.Recompiler.recompile(Recompiler.java:376)
	at org.apache.sysml.hops.recompile.Recompiler.recompileHopsDag(Recompiler.java:163)
	at org.apache.sysml.hops.codegen.SpoofCompiler.generateCodeFromHopDAGsToInst(SpoofCompiler.java:315)
	at org.apache.sysml.hops.codegen.SpoofCompiler.generateCodeFromProgramBlock(SpoofCompiler.java:294)
	at org.apache.sysml.hops.codegen.SpoofCompiler.generateCodeFromProgramBlock(SpoofCompiler.java:274)
	at org.apache.sysml.hops.codegen.SpoofCompiler.generateCode(SpoofCompiler.java:200)
	at org.apache.sysml.parser.DMLTranslator.codgenHopsDAG(DMLTranslator.java:308)
	at org.apache.sysml.parser.DMLTranslator.getRuntimeProgram(DMLTranslator.java:451)
	at org.apache.sysml.api.DMLScript.execute(DMLScript.java:711)
	at org.apache.sysml.api.DMLScript.executeScript(DMLScript.java:511)
	at org.apache.sysml.api.DMLScript.main(DMLScript.java:238)
	at org.apache.sysml.test.integration.AutomatedTestBase.runTest(AutomatedTestBase.java:1222)
	at org.apache.sysml.test.integration.functions.codegenalg.AlgorithmKMeans.runKMeansTest(AlgorithmKMeans.java:341)
	at org.apache.sysml.test.integration.functions.codegenalg.AlgorithmKMeans.testKMeansSparseMulSingleCPFuseAll(AlgorithmKMeans.java:204)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: org.apache.sysml.runtime.DMLRuntimeException: Failed to compile class codegen.TMP78.
	at org.apache.sysml.runtime.codegen.CodegenUtils.compileClassJanino(CodegenUtils.java:197)
	at org.apache.sysml.runtime.codegen.CodegenUtils.compileClass(CodegenUtils.java:82)
	at org.apache.sysml.hops.codegen.SpoofCompiler.optimize(SpoofCompiler.java:416)
	... 38 more
Caused by: org.codehaus.commons.compiler.CompileException: Line 14, Column 38: Binary numeric promotion not possible on types "double[]" and "double"
	at org.codehaus.janino.UnitCompiler.compileError(UnitCompiler.java:11004)
	at org.codehaus.janino.UnitCompiler.binaryNumericPromotionType(UnitCompiler.java:10237)
	at org.codehaus.janino.UnitCompiler.binaryNumericPromotion(UnitCompiler.java:10206)
	at org.codehaus.janino.UnitCompiler.binaryNumericPromotion(UnitCompiler.java:10180)
	at org.codehaus.janino.UnitCompiler.compileArithmeticOperation(UnitCompiler.java:6865)
	at org.codehaus.janino.UnitCompiler.compileGet2(UnitCompiler.java:4367)
	at org.codehaus.janino.UnitCompiler.access$6900(UnitCompiler.java:206)
	at org.codehaus.janino.UnitCompiler$12.visitBinaryOperation(UnitCompiler.java:3768)
	at org.codehaus.janino.UnitCompiler$12.visitBinaryOperation(UnitCompiler.java:3762)
	at org.codehaus.janino.Java$BinaryOperation.accept(Java.java:4155)
	at org.codehaus.janino.UnitCompiler.compileGet(UnitCompiler.java:3762)
	at org.codehaus.janino.UnitCompiler.compileGetValue(UnitCompiler.java:4933)
	at org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:2330)
	at org.codehaus.janino.UnitCompiler.access$2600(UnitCompiler.java:206)
	at org.codehaus.janino.UnitCompiler$6.visitLocalVariableDeclarationStatement(UnitCompiler.java:1386)
	at org.codehaus.janino.UnitCompiler$6.visitLocalVariableDeclarationStatement(UnitCompiler.java:1370)
	at org.codehaus.janino.Java$LocalVariableDeclarationStatement.accept(Java.java:2974)
	at org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:1370)
	at org.codehaus.janino.UnitCompiler.compileStatements(UnitCompiler.java:1450)
	at org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:2811)
	at org.codehaus.janino.UnitCompiler.compileDeclaredMethods(UnitCompiler.java:1262)
	at org.codehaus.janino.UnitCompiler.compileDeclaredMethods(UnitCompiler.java:1234)
	at org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:538)
	at org.codehaus.janino.UnitCompiler.compile2(UnitCompiler.java:420)
	at org.codehaus.janino.UnitCompiler.access$400(UnitCompiler.java:206)
	at org.codehaus.janino.UnitCompiler$2.visitPackageMemberClassDeclaration(UnitCompiler.java:374)
	at org.codehaus.janino.UnitCompiler$2.visitPackageMemberClassDeclaration(UnitCompiler.java:369)
	at org.codehaus.janino.Java$AbstractPackageMemberClassDeclaration.accept(Java.java:1309)
	at org.codehaus.janino.UnitCompiler.compile(UnitCompiler.java:369)
	at org.codehaus.janino.UnitCompiler.compileUnit(UnitCompiler.java:345)
	at org.codehaus.janino.SimpleCompiler.compileToClassLoader(SimpleCompiler.java:396)
	at org.codehaus.janino.SimpleCompiler.cook(SimpleCompiler.java:205)
	at org.codehaus.janino.SimpleCompiler.cook(SimpleCompiler.java:196)
	at org.codehaus.commons.compiler.Cookable.cook(Cookable.java:91)
	at org.codehaus.commons.compiler.Cookable.cook(Cookable.java:84)
	at org.apache.sysml.runtime.codegen.CodegenUtils.compileClassJanino(CodegenUtils.java:186)
	... 40 more
org.apache.sysml.api.DMLException: org.apache.sysml.runtime.DMLRuntimeException: org.apache.sysml.runtime.DMLRuntimeException: Failed to compile class codegen.TMP78.
	at org.apache.sysml.api.DMLScript.executeScript(DMLScript.java:538)
	at org.apache.sysml.api.DMLScript.main(DMLScript.java:238)
	at org.apache.sysml.test.integration.AutomatedTestBase.runTest(AutomatedTestBase.java:1222)
	at org.apache.sysml.test.integration.functions.codegenalg.AlgorithmKMeans.runKMeansTest(AlgorithmKMeans.java:341)`